hawkbit-core/src/main/java/org/eclipse/hawkbit/exception/ErrorHandlerChain.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (c) 2021 Bosch.IO GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.exception;
+
+/**
+ * A chain where an error is passed through all handlers until one is found that
+ * can process the error.
+ *
+ * @param <T>
+ *            generic type of event.
+ */
+public interface ErrorHandlerChain<T> {
```

Comment:
maybe we don't need this interface?
seems to me like there will not be more than one implementation of the chain, right?

Reply:
As discussed, we will be having two implementations, hence will retain the interface